cmd/tailscale/cli/serve_v2.go

```diff
@@ -818,7 +818,7 @@ func srvTypeAndPortFromFlags(e *serveEnv) (srvType serveType, srvPort uint16, is
 		srvPort = 443
 		defaultFlags = true
 	}
-	return srvType, srvPort, defaultFlags && !e.service.IsEmpty(), nil
+	return srvType, srvPort, defaultFlags && e.service != "", nil
```

Comment:
optional: if you think it looks nicer, `e.service != noService` will work here instead. It's equivalent, use whichever you like better.